exo/networking/grpc/grpc_discovery.py

```diff
@@ -34,6 +35,16 @@ def __init__(self, node_id: str, node_port: int, listen_port: int, broadcast_por
         self.broadcast_task = None
         self.listen_task = None
         self.cleanup_task = None
+        self.interfaces = self.get_network_interfaces()
```

Comment:
If we can remove`self.interfaces` in `task_listen_for_peers` (https://github.com/exo-explore/exo/pull/47/files#r1685535090) then we should move this out of the constructor to `task_broadcas_presence` and have it be error handled (in each loop, we can check if `self.interfaces` is already set and if not, set it)

Reply:
I'll get on with this.

Reply:
- The `task_listen_for_peers` method now listens on all interfaces using '0.0.0.0'.
- The `self.interfaces` attribute is removed from the constructor.
- The `task_broadcast_presence` method now updates the list of interfaces in each iteration, ensuring it always uses the most up-to-date list of network interfaces.